src/main/java/io/gravitee/resource/schema_registry/api/Schema.java

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.resource.schema_registry.api;
+
+public interface Schema {
```

Comment:
question: In addition of the content, should we provide also:
* the schema type (JSON/Avro/Protobuf)
* the version
* the id or subject
I'm thinking of the usage of this schema in a policy where we may have to do some checks.
wdyt?

Reply:
Regarding the type, the registry does not give any information on it (at least I didn't found). We could eventually "guess" based on the content.
We can add the `id`, the subject and the version

Jeoffrey also suggests we add a `transform` method that could handle data transformation based on the current schema. The idea would be to have a specific implementation class for each kind of schema, starting with Avro.  I planned to work on that after ensuring I had a simple working implementation.

Reply:
ok, in that case should we merge on alpha branch to avoid releasing a 1.0.0 before we are confident in the final interface ?

Reply:
🤷  It is just a version number. I don't see why it is so bad to create version but ok I will create an alpha branch ;)